cardinal/ecs/world.go

```diff
@@ -38,7 +42,8 @@ type StorageAccessor struct {
 type World struct {
 	namespace                Namespace
 	store                    storage.WorldStorage
-	systems                  []System
+	systems                  []RegisteredSystem
+	systemNames              []string
```

Comment:
systemNames is required because once a System is curried into RegisteredSystem, the function name is lost and reflection can no longer retrieve it. This is required for printing system names short of a full refactoring of the concept of a System. 

Of course if the user used an anonymous function, the name will not be known no matter what.